### ISSUE TITLE
Speed up the GitHub actions

### ISSUE THREAD

**Issue by tms-bananaquit** [mitre/menelaus#6]
- Can we get a performance improvement by using something other than ubuntu-latest?
- Can we get a performance improvement by using docker images, instead of repeatedly configuring environments?
- The pipeline currently complains about not having wheel. Speed up by installing it, or by using docker as above?
- Is it faster to use e.g. `venv` over `conda`?

**Comment by Anmol-Srivastava** [comment]
@tms-bananaquit initial findings

1. Unclear whether alternatives to `ubuntu-latest` are meaningfully faster; further, `sklearn` and `tensorflow` are just some repositories that use `ubuntu-latest` in their files.
2. [It doesn't seem like Docker images themselves provide significant advantage](https://evilmartians.com/chronicles/build-images-on-github-actions-with-docker-layer-caching), but rather that any possible caching is where the speedups would come through. Other packages also don't use Docker steps in their builds. 
3. **_TBD, looking into this_**
4. I'm not sure about this -- within Actions no real virtual environment is used (or, I feel, needed). Later on, there may be a need to support different versions of Python, in which case we may experiment with e.g. `venv` + a tool like [`pyenv`](https://github.com/pyenv/pyenv), but I'm omitting this for the time being.

**Comment by tms-bananaquit** [comment]
Looks like (3) is a non-issue for now.
